packages/cpp/src/common/cpp-preferences.ts

```diff
@@ -14,32 +14,38 @@ import {
     PreferenceSchema
 } from '@theia/preferences/lib/common';
 
+export const CLANGD_COMMAND_DEFAULT = 'clangd';
+
 export const CppConfigSchema: PreferenceSchema = {
     "type": "object",
     "properties": {
-        "cpp.clangdCompileCommandsPath": {
+        "cpp.clangdCompilationDatabaseDirectory": {
```

Comment:
Isn't this information needed when ever a clangd is used? If yes, it should not be a theia-specific property but clangd specific, such that it works in other IDEs using clangd, too.

Reply:
By default, it will look in parent folders to find it. But often, it will be in a build folder inside the workspace. In fact, this should be user-specific as users can switch configuration (debug, release, different target) as they see fit, even while clangd is running (see https://github.com/theia-ide/theia/pull/780). From what I understand, there would be only one user per backend and per workspace, right? The user needs to be able to set the configuration (including the compilation database) both during initialization (done with a preference here) and at runtime without affecting the other users.